### PR TITLE
pre_defined_classes-simple_programs_taking_string_user_input-medium-q8

### PR DESCRIPTION
**Pull request by yuqi88**
## PR Checklist
- [ ]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [ ]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [ ]  Readable (can a student understand what’s wrong?)
    - [ ]  Not too feed-y (does it give away the answer?)
    - [ ]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [ ]  Specific (does the fail message relate to the question?)
- [ ]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [ ]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [ ]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

**Comment by yuqi88** [comment]
Changed the question to deal with `new RandomChar( 'A', (char) ('Z' + 1), "ranChar1")` and that 'Z'+1 in the ASCII table is [.

**Comment by yuqi88** [comment]
Issue with RandomChar( ) clause:
The test will not pass given `System.lineSeparator()` as the input.
This might be something @Hedgemon4 you want to look into. Seems like it's a similar issue as the RandomInteger( ).
#222 

For now, I'll add a clarification to the question so that no-value input is not allowed.

image

**Comment by Hedgemon4** [comment]
Waiting for https://github.com/canvas-gamification/canvas-gamification-junit-tests/issues/272

**Comment by Opeyem1a** [comment]
Should be fixed once framework changes are pushed 👍🏾

File: src/pre_defined_classes/simple_programs_taking_string_user_input/medium/q8/UserName.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package pre_defined_classes.simple_programs_taking_string_user_input.medium.q8;
+
+/*
+When you create a new account on a website, you get to choose your desired username. If it is not available, some
+suggestions will be made. Make a program that asks for your desired username and then suggests an alternative username
+by adding three random letters from A to Z (including both ends) to the original desired username. Assume all usernames
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_string_user_input/medium/q8/UserName.java:6]:
```suggestion
by adding three random letters from A to Z (inclusive) to the original desired username. Assume all usernames
```

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**